enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTestLogic.java

```diff
@@ -875,4 +921,49 @@ public void shouldSetCorrectMultiRolePermissions() throws Exception
         testFailCreateUser( schemaSubject, PERMISSION_DENIED );
         assertEmpty( schemaSubject, "CALL dbms.changePassword( '321' )" );
     }
+
+    // --------------------- helpers -----------------------
+
+    private void shouldTerminateTransactionsForUser( S subject, String procedure ) throws Throwable
+    {
+        ThreadedTransactionCreate<S> userThread = new ThreadedTransactionCreate<>( neo );
+        userThread.execute( threading, subject );
+        userThread.barrier.await();
+
+        assertEmpty( adminSubject, "CALL " + String.format(procedure, neo.nameOf( subject ) ) );
+
+        assertSuccess( adminSubject, "CALL dbms.listTransactions()",
+                r -> assertKeyIsMap( r, "username", "activeTransactions", map( "adminSubject", "1" ) ) );
+
+        userThread.closeAndAssertTransactionTermination();
+
+        assertEmpty( adminSubject, "MATCH (n:Test) RETURN n.name AS name" );
+    }
+
+    private void shouldTerminateSessionsForUser( String username, String password, String procedure ) throws Exception
+    {
+        Connection boltConnection = new SocketConnection();
+        HostnamePort boltAddress = new HostnamePort( "localhost:7687" );
+        authenticate( boltConnection, boltAddress, username, password );
```

Comment:
It feels a bit weird with a Bolt client session for the embedded interaction. Could this be abstracted into NeoInteractionLevel and maybe be a no-op in NeoShallowEmbeddedInteraction? It should be enough to test with the client-server interaction levels (rest and bolt)